src/sinks/hdfs/mod.rs

```diff
@@ -0,0 +1,75 @@
+//! `hdfs` sink.
+//!
+//! This sink will send it's output to hdfs.
+//!
+//! `hdfs` is an opendal based services. This mod itself only provide
+//! config to build an [`OpendalSink`]. All real implement are powered by
+//! [`OpendalSink`].
+
+use super::opendal_common::*;
+use super::Healthcheck;
+use crate::config::{GenerateConfig, SinkConfig, SinkContext};
+use opendal::services::Hdfs;
+use opendal::Operator;
+use vector_config::configurable_component;
+use vector_core::{
+    config::{AcknowledgementsConfig, Input},
+    sink::VectorSink,
+};
+
+/// A sink that dumps its output to hdfs.
+#[configurable_component(sink("hdfs"))]
+#[derive(Clone, Debug)]
```

Comment:
```suggestion
#[derive(Clone, Debug)]
#[serde(deny_unknown_fields)]
```

**suggestion**: Provides a better UX by defending against typos in option names.